src/components/boardComponents/useItemActions.jsx

```diff
@@ -89,8 +90,49 @@ export const useItemActions = () => {
     updateAvailableActions();
   }, [selectedItems, updateAvailableActions]);
 
-  // Align selection to center
-  const align = useRecoilCallback(
+// Stack selection to Center
+const stackToCenter = useRecoilCallback(
+  ({ snapshot }) => async () => {
+    const selectedItemList = await getSelectedItemList(snapshot);
+
+    // Compute
+    const minMax = { min: {}, max: {} };
+    minMax.min.x = Math.min(...selectedItemList.map(({ x }) => x));
+    minMax.min.y = Math.min(...selectedItemList.map(({ y }) => y));
+    minMax.max.x = Math.max(
+      ...selectedItemList.map(
+        ({ x, id }) => x + document.getElementById(id).clientWidth
+      )
+    );
+    minMax.max.y = Math.max(
+      ...selectedItemList.map(
+        ({ y, id }) => y + document.getElementById(id).clientHeight
+      )
+    );
+
+    const [newX, newY] = [minMax.min.x + (minMax.max.x - minMax.min.x) / 2 ,  minMax.min.y + (minMax.max.y - minMax.min.y) / 2];
+    let stackThickness = 0;
+      if (selectedItemList.length >= 32) {
+          stackThickness = 0.5;
+        } else {
+          stackThickness = 1;
+        }
+    let index = -1;
+    batchUpdateItems(selectedItems, (item) => {
+      index += 1;
+      const { clientWidth, clientHeight } = document.getElementById(item.id);
+      return {
+        ...item,
+        x: newX - clientWidth / 2 + index * stackThickness,
+        y: newY - clientHeight / 2 + (selectedItemList.length - index - 1) * stackThickness,
+      };
+    });
+  },
+  [getSelectedItemList, batchUpdateItems, selectedItems]
+);
```

Comment:
To complete my last comment, this is the partial fix I propose:

```suggestion
const stackToCenter = useRecoilCallback(
  ({ snapshot }) => async () => {
    const selectedItemList = await getSelectedItemList(snapshot);

    // Should be the center here -> the first card should not move!
    // Can be the exact same computation for other align actions
    const [newX, newY] = [selectedItemList[0].x, selectedItemList[0].y];
    
    let stackThickness = 0;
    if (selectedItemList.length >= 32) {
        stackThickness = 0.5;
    } else {
        stackThickness = 1;
    }
    let index = -1;
    batchUpdateItems(selectedItems, (item) => {
      index += 1;
      const { clientWidth, clientHeight } = document.getElementById(item.id);
      return {
        ...item,
        x: newX - clientWidth / 2 + index * stackThickness,
        y: newY - clientHeight / 2 + (selectedItemList.length - index - 1) * stackThickness,
      };
    });
  },
  [getSelectedItemList, batchUpdateItems, selectedItems]
);
```

Reply:
I think we can use the same origin item for all alignments, this way you keep stability and its seems more intuitive to me.
 
- For the stack, all cards are stacked on top of the first one.
- For the line the first item in `selectedItemList` is the most at the left then the second just after, ... it's like "carressing" a deck of card to allow payer to pick one (Like in this video https://youtu.be/0ytlTLoKknU?t=121)
- For the square, the same, the first item in the `selectedItemList` is the top left card then the second just after, etc...

If you switch from one alignement to another, the first card is the same and never move, so we have the stability even if you click multiple times

One use case, for example, is when you want to put back some cards in a deck you select all card and click stack action. The result should be that only previously discarded cards move and the other stays.

Reply:
Ok, it's clearer to me, I understand now.

Reply:
Effectively by using the first element of the list to carry out the stacks and the alignments, it is simpler and clearer.
I stabilized the displacement effect of the stack to center function.

I made the changes here: https://github.com/jrmi/airboardgame/pull/282/commits/8353a0b71b06dab64d3b5571e3e6c85eaa0adc1a

A video to illustrate:
https://youtu.be/_FUZJrGg5PM


I noticed two bugs that I show in two other videos:
https://youtu.be/nuk4UJTm7zY
https://youtu.be/ALtS6JRxhaE

Reply:
Nice, it's really cool. Unless you need it, you can remove the stack to center action to keep only yours now.

For the first bug, you have to know that the order of the cards in the item stack is reversed when you reveal the card to simulate the effect of having multiple cards in hand and reveal them: the one on bottom is the first you see. So the alignment is now from the one which was on top before. I don't think it should be a problem thought, but as I see in your video, if you click again it should keep stability and it doesn't... we have to investigate. 

For the second bug, I already noticed it but don't succeed to reproduce it in a predictive way. Your video will help me to do it, thanks. I already have a clue of where I need to look for.

Reply:
Note that this bug only occurs with the "Reveal / Return" action and not with the "Return for me" action.
I did not understand why reverse the order?
Is it really necessary? Because it could be a source of bugs.

For the 2nd bug, I explain in the description of the video how to reproduce it.


For the "stackToCenter" action, are you sure you don't want to keep it?
If you don't want to, then here are some suggestions:
- we could keep the function in "useItemActions.jsx" (as it is coded, could it be used one day?)
- we keep the name of the function which is explicit "stackToTopLeft",
- you can change the "Stack To Top Left" label to "Stack".

Reply:
> Note that this bug only occurs with the "Reveal / Return" action and not with the "Return for me" action.
> I did not understand why reverse the order?
> Is it really necessary? Because it could be a source of bugs.

Sometimes in some games you return many cards at same times and their should be reversed. For example, in Solitaire games, you return the cards 3 by 3 and only the most on the top is available until you place it. Actually, that's the natural behaviour when you have real cards in your hands and I want that Airboargame mimic the natural behaviour as much as possible to be able to support real board games without rules or behaviours modification.

However, If you have a specific use case in mind where it can cause trouble, share it and we'll consider adding a non reversing reveal action (by adding a parameter to the function). The more we discuss, the more I think the ticket #317 should solve a lot of use case (but also make game creation a bit more complex ;-) ).

What do you think?

Reply:
> For the "stackToCenter" action, are you sure you don't want to keep it?
> If you don't want to, then here are some suggestions:
> 
>     * we could keep the function in "useItemActions.jsx" (as it is coded, could it be used one day?)
> 
>     * we keep the name of the function which is explicit "stackToTopLeft",
> 
>     * you can change the "Stack To Top Left" label to "Stack".

Ok with that!

Reply:
> 
> 
> > Note that this bug only occurs with the "Reveal / Return" action and not with the "Return for me" action.
> > I did not understand why reverse the order?
> > Is it really necessary? Because it could be a source of bugs.
> 
> Sometimes in some games you return many cards at same times and their should be reversed. For example, in Solitaire games, you return the cards 3 by 3 and only the most on the top is available until you place it. Actually, that's the natural behaviour when you have real cards in your hands and I want that Airboargame mimic the natural behaviour as much as possible to be able to support real board games without rules or behaviours modification.
> 
> However, If you have a specific use case in mind where it can cause trouble, share it and we'll consider adding a non reversing reveal action (by adding a parameter to the function). The more we discuss, the more I think the ticket #317 should solve a lot of use case (but also make game creation a bit more complex ;-) ).
> 
> What do you think?

I think there are many use case when also we need to keep order. For example, when we want to return our cards in hand, the order don't change.

The both case can be natural :
- by rotation of the hand,  we can horizontally return  several cards at the same time : the order are inverted.
- by rotation of the hand, we can vertically return several cards at the same time: the order are not inverted.

 The ticket #317 should  make game creation a bit more complex (more possibilities), but not necessary difficult :-p

Reply:
Fine, keep both actions ;-)

Reply:
You also can do it in another PR if you prefer to have this PR live earlier.